projects/plugins/jetpack/_inc/lib/class-jetpack-podcast-helper.php

```diff
@@ -98,14 +98,21 @@ function ( $track ) {
 	/**
 	 * Gets a specific track from the supplied feed URL.
 	 *
-	 * @param string $guid     The GUID of the track.
-	 * @return array|WP_Error  The track object or an error object.
+	 * @param string $guid          The GUID of the track.
+	 * @param bool   $force_refresh Whether to force a refresh.
+	 * @return array|WP_Error       The track object or an error object.
 	 */
-	public function get_track_data( $guid ) {
+	public function get_track_data( $guid, $force_refresh = false ) {
 		// Try loading track data from the cache.
 		$transient_key = 'jetpack_podcast_' . md5( "$this->feed::$guid" );
 		$track_data    = get_transient( $transient_key );
 
+		// Clear the cache if force_refresh param is set.
+		if ( true === $force_refresh ) {
+			delete_transient( $track_data );
+			add_action( 'wp_feed_options', array( __CLASS__, 'reset_simplepie_cache' ) );
+		}
+
```

Comment:
I think this block needs to be placed before setting `$track_data` from cache. Otherwise the cache could still used.

Reply:
Addressed here: https://github.com/Automattic/jetpack/pull/18573/commits/eb9b550f8ad3d21ea9b099e94fdd7aad9b69192d I think I was also calling `delete_transient` on the value instead of the key.